text/1071-cell.md

```diff
@@ -0,0 +1,296 @@
+---
+stage: accepted
+start-date: 2025-01-19T00:00:00.000Z
+release-date: # In format YYYY-MM-DDT00:00:00.000Z
+release-versions:
+teams: # delete teams that aren't relevant
+  - cli
+  - data
+  - framework
+  - learning
+  - steering
+  - typescript
+prs:
+  accepted: https://github.com/emberjs/rfcs/pull/1071
+project-link:
+suite: 
+---
+
+<!--- 
+Directions for above: 
+
+stage: Leave as is
+start-date: Fill in with today's date, 2032-12-01T00:00:00.000Z
+release-date: Leave as is
+release-versions: Leave as is
+teams: Include only the [team(s)](README.md#relevant-teams) for which this RFC applies
+prs:
+  accepted: Fill this in with the URL for the Proposal RFC PR
+project-link: Leave as is
+suite: Leave as is
+-->
+
+<-- Replace "RFC title" with the title of your RFC -->
+# Introduce `Cell` 
+
+## Summary
+
+This RFC introduces a new tracking primitive, which represents a single value, the `Cell`.
+
+## Motivation
+
+The `Cell` is part of the "spreadsheet analogy" when talking about reactivity -- it represents a single tracked value, and can be created without the use of a class, making it a primate candidate for demos[^demos] and for creating reactive values in function-based APIs, such as _helpers_, _modifiers_, or _resources_. They also provide a benefit in testing as well, since tests tend to want to work with some state, the `Cell` is wholly encapsulated, and can be quickly created with 0 ceremony. 
+
+This is not too dissimilar to the [Tracked Storage Primitive in RFC#669](https://github.com/emberjs/rfcs/blob/master/text/0669-tracked-storage-primitive.md). The `Cell` provides more ergonomic benefits as it doesn't require 3 imports to use. 
+
+The `Cell` was prototyped in [Starbeam](https://starbeamjs.com/guides/fundamentals/cells.html) and has been available for folks to try out in ember via [ember-resources](https://github.com/NullVoxPopuli/ember-resources/tree/main/docs/docs). 
+
+[^demos]: demos _must_ over simplify to bring attention to a specific concept. Too much syntax getting in the way easily distracts from what is trying to be demoed. This has benefits for actual app development as well though, as we're, by focusing on concise demo-ability, gradually removing the amount of typing needed to create features. 
+
+## Detailed design
+
+
+### Types 
+
+Some interfaces to share with future low-level reactive primitives:
+
+```ts
+
+interface Reactive<Value> {
+    /**
+    * The underlying value
+    */
+    current: Value;
+    /**
+    * Returns the underlying value
+    */
+    read(): Value;
```

Comment:
Could you explain why we need explicit `read` (and `set`) APIs? Do we actually?

I understand why other ecosystems like Solid that (for whatever legitimate or stupid reason) avoid using classes have more explicit consume/set APIs, but (as we are not afraid of OOP here) couldn't these just be getters and setters on `current`?

Reply:
I think this is a good point to address. `current` is better to consume from templates, so perhaps we drop `read()`?

Reply:
I forgot to respond to this thread, but `read()` is useful for when you need to entangle but not use the value.

This may be approaching anti-pattern, but the reality is that in real apps, some folks do:
```js
get foo() {
  this.otherThing; // tracked, consume it
  // ...
}
```
and eslint (appropriately) lints against that.

the above would be re-written as:
```js
get foo() {
  this.someCell.read()
  // ...
}
```
which _looks like_ a side-effect, which appropriately should give folks pause, but moreso understanding about what's going on and why -- it looks more intentional, as methods are generally accepted to _maybe_ have a side-effect (the side-effect in this case is entangling)

Reply:
Calling `.read()` without returning or assigning looks pretty much as awkward as the other example to me. I feel both examples deserve a comment explaining with this is being done (as it looks like you say a bit like an anti-pattern), so you could just as well write that comment following `// eslint-disable-line`.  If making ESLint shut up is the only reason to have this, I would say that's not worth it.

Reply:
I added an example that is better than the above. "Side-signals", based off some disco with @runspired